web/src/store/pipelines.ts

```diff
@@ -1,99 +1,103 @@
+    if (!pipeline.steps) {
+      pipeline.steps = [];
+    }
 
-      this.pipelines = {
-        ...this.pipelines,
-        [_repoSlug]: repoPipelines,
-      };
-    },
-    setStep(owner: string, repo: string, pipelineNumber: number, step: PipelineStep) {
-      const pipeline = this.getPipeline(ref(owner), ref(repo), ref(pipelineNumber.toString())).value;
-      if (!pipeline) {
-        throw new Error("Can't find pipeline");
-      }
+    pipeline.steps = [...pipeline.steps.filter((p) => p.pid !== step.pid), step];
+    setPipeline(owner, repo, pipeline);
+  }
 
-      if (!pipeline.steps) {
-        pipeline.steps = [];
-      }
+  async function loadRepoPipelines(owner: string, repo: string) {
+    const _pipelines = await apiClient.getPipelineList(owner, repo);
+    _pipelines.forEach((pipeline) => {
+      setPipeline(owner, repo, pipeline);
+    });
+  }
 
-      pipeline.steps = [...pipeline.steps.filter((p) => p.pid !== step.pid), step];
-      this.setPipeline(owner, repo, pipeline);
-    },
-    setPipelineFeedItem(pipeline: PipelineFeed) {
-      const pipelineFeed = this.pipelineFeed.filter((b) => b.id !== pipeline.id);
-      this.pipelineFeed = [...pipelineFeed, pipeline];
-    },
+  async function loadPipeline(owner: string, repo: string, pipelinesNumber: number) {
+    const pipeline = await apiClient.getPipeline(owner, repo, pipelinesNumber);
+    setPipeline(owner, repo, pipeline);
+  }
 
-    // getters
-    getPipelines(owner: Ref<string>, repo: Ref<string>) {
-      return computed(() => {
-        const slug = repoSlug(owner.value, repo.value);
-        return toRef(this.pipelines, slug).value;
-      });
-    },
-    getSortedPipelines(owner: Ref<string>, repo: Ref<string>) {
-      return computed(() => Object.values(this.getPipelines(owner, repo).value || []).sort(comparePipelines));
-    },
-    getActivePipelines(owner: Ref<string>, repo: Ref<string>) {
-      const pipelines = this.getPipelines(owner, repo);
-      return computed(() => Object.values(pipelines.value).filter(isPipelineActive));
-    },
-    getPipeline(owner: Ref<string>, repo: Ref<string>, pipelineNumber: Ref<string>) {
-      const pipelines = this.getPipelines(owner, repo);
-      return computed(() => (pipelines.value || {})[parseInt(pipelineNumber.value, 10)]);
-    },
+  const pipelineFeed = computed(() =>
+    Array.from(pipelines.entries())
+      .reduce((acc, [_repoSlug, repoPipelines]) => {
+        const repoPipelinesArray = Array.from(repoPipelines.entries()).map(
+          ([_pipelineNumber, pipeline]) =>
+            <PipelineFeed>{
+              ...pipeline,
+              full_name: _repoSlug,
+              owner: _repoSlug.split('/')[0],
+              name: _repoSlug.split('/')[1],
+              number: _pipelineNumber,
+            },
+        );
+        return [...acc, ...repoPipelinesArray];
+      }, [] as PipelineFeed[])
```

Comment:
@anbraten Can't you pass `PipelineFeed[]` as type parameter to `reduce`?

Reply:
changed it